runtime/lua/vim/_editor.lua

```diff
@@ -1042,50 +1042,75 @@ end
 ---
 ---@return string|nil # Deprecated message, or nil if no message was shown.
 function vim.deprecate(name, alternative, version, plugin, backtrace)
-  plugin = plugin or 'Nvim'
-  local will_be_removed = 'will be removed'
-
-  -- Only issue warning if feature is hard-deprecated as specified by MAINTAIN.md.
-  -- Example: if removal_version is 0.12 (soft-deprecated since 0.10-dev), show warnings starting at
-  -- 0.11, including 0.11-dev
-  if plugin == 'Nvim' then
-    local major, minor = version:match('(%d+)%.(%d+)')
-    major, minor = tonumber(major), tonumber(minor)
-
-    local hard_deprecated_since = string.format('nvim-%d.%d', major, minor - 1)
-    -- Assume there will be no next minor version before bumping up the major version
-    local is_hard_deprecated = minor == 0 or vim.fn.has(hard_deprecated_since) == 1
-    if not is_hard_deprecated then
-      return
-    end
-
-    local removal_version = string.format('nvim-%d.%d', major, minor)
-    will_be_removed = vim.fn.has(removal_version) == 1 and 'was removed' or will_be_removed
-  else
-    vim.validate {
-      name = { name, 'string' },
-      alternative = { alternative, 'string', true },
-      version = { version, 'string', true },
-      plugin = { plugin, 'string', true },
-    }
-  end
+  vim.validate {
+    name = { name, 'string' },
+    alternative = { alternative, 'string', true },
+    version = { version, 'string', true },
+    plugin = { plugin, 'string', true },
+  }
 
   local msg = ('%s is deprecated'):format(name)
   msg = alternative and ('%s, use %s instead.'):format(msg, alternative) or (msg .. '.')
-  msg = ('%s%s\nFeature %s in %s %s'):format(
-    msg,
-    (plugin == 'Nvim' and ' :help deprecated' or ''),
-    will_be_removed,
-    plugin,
-    version
-  )
+  msg = ('%s\nFeature will be removed in %s %s'):format(msg, plugin, version)
   local displayed = vim.notify_once(msg, vim.log.levels.WARN)
   if displayed and backtrace ~= false then
     vim.notify(debug.traceback('', 2):sub(2), vim.log.levels.WARN)
   end
   return displayed and msg or nil
 end
 
+do
+  local function truncated_echo(msg)
+    -- Truncate message to avoid hit-enter-prompt
+    local max_width = vim.o.columns * math.max(vim.o.cmdheight - 1, 0) + vim.v.echospace
+    local msg_truncated = string.sub(msg, 1, max_width)
+    vim.api.nvim_echo({ { msg_truncated, 'WarningMsg' } }, true, {})
+  end
+
+  local notified = false
+
+  function vim._notify_once_truncated(msg)
+    if not notified then
```

Comment:
Unlike the semantic of `vim.notify_once` where only one notification is made per `msg`, this will suppress echoing for *any* further message. I can get why you chose this way --- to prevent multiple messages printed during startup and hence the 'hit-enter-prompt' (even if there are 2+ different messages). (Also, nitpicking on name: it's not using "vim.notify").

However, there is one problem about this. `vim._deprecate` warnings (for Nvim Lua APIs) *can* still happen again after vim startup and a warning message is shown once, when a deprecated function is used in later calls by plugin or user config upon some events, scheduled asynchronously, etc. Those warnings will be silenced, although collected in `:checkhealth vim.deprecate`.

Reply:
I don't get it. vim.notify triggers the press enter prompt. How does using that help?

Reply:
Hmm actually you're right. I was somehow assuming users would be overriding `vim.notify` with [some plugins](https://github.com/rcarriga/nvim-notify) so that it doesn't write into `:message`. But if such plugins are not used, i.e. with the factory default setting of `vim.notify`, it will trigger the hit-enter-prompt (no matter vim has entered or not). Never mind about the potential solution suggested. If we'd want to completely eliminate the hit-enter-prompt, we shouldn't use `vim.notify` at all.

Reply:
On the other hand a non-blocking notification likr the plugin could also work, although I suspect it'll be more ambitious.